src/cse-machine/operators.ts

```diff
@@ -0,0 +1,473 @@
+
+                if (identifier === '>') {
+                    ret_value = compare_res > 0;
+                } else if(identifier === '>=') {
+                    ret_value = compare_res >= 0;
+                } else if(identifier === '<') {
+                    ret_value = compare_res < 0;
+                } else if(identifier === '<=') {
+                    ret_value = compare_res <= 0;
+                } else if(identifier === '===') {
+                    ret_value = compare_res === 0;
+                } else if(identifier === '!==') {
+                    ret_value = compare_res !== 0;
+                } else {
+                    // TODO: error
+                }
+
+                
+            } else {
+                if (identifier === '>') {
+                    ret_value = left.value > right.value;
+                } else if(identifier === '>=') {
+                    ret_value = left.value >= right.value;
+                } else if(identifier === '<') {
+                    ret_value = left.value < right.value;
+                } else if(identifier === '<=') {
+                    ret_value = left.value <= right.value;
+                } else if(identifier === '===') {
+                    ret_value = left.value === right.value;
+                } else if(identifier === '!==') {
+                    ret_value = left.value !== right.value;
+                } else {
+                    // TODO: error
+                }
+            }
+
+            
+        }
+
+        return {
+            type: ret_type,
+            value: ret_value
+        };
+    }
+}
+
+/**
+ * pyCompare: Compares a Python-style big integer (int_num) with a float (float_num),
+ * returning -1, 0, or 1 for less-than, equal, or greater-than.
+ * 
+ * This logic follows CPython's approach in floatobject.c, ensuring Python-like semantics:
+ * 
+ * 1. Special Values:
+ *    - If float_num is inf, any finite int_num is smaller (returns -1).
+ *    - If float_num is -inf, any finite int_num is larger (returns 1).
+ * 
+ * 2. Compare by Sign:
+ *    - Determine each number’s sign (negative, zero, or positive). If they differ, return based on sign.
+ *    - If both are zero, treat them as equal.
+ * 
+ * 3. Safe Conversion:
+ *    - If |int_num| <= 2^53, safely convert it to a double and do a normal floating comparison.
+ * 
+ * 4. Handling Large Integers:
+ *    - For int_num beyond 2^53, approximate the magnitudes via exponent/bit length.
+ *    - Compare the integer’s digit count with float_num’s order of magnitude.
+ * 
+ * 5. Close Cases:
+ *    - If both integer and float have the same digit count, convert float_num to a “big-int-like” string
+ *      (approximateBigIntString) and compare lexicographically to int_num’s string.
+ * 
+ * By layering sign checks, safe numeric range checks, and approximate comparisons,
+ * we achieve a Python-like ordering of large integers vs floats.
+ */
+
+function pyCompare(int_num : any, float_num : any) {
```

Comment:
Instead of having a single operator, we should follow Python convention and call `__eq__` `__lt__` or `__gt__` and such, then each type will have the comparison overloaded. However I understand you won't have time to do that in this PR. So please add a comment somewhere saying that.